source/state_representation/include/state_representation/space/joint/JointState.hpp

```diff
@@ -642,15 +642,14 @@ inline void JointState::set_all_state_variables(const Eigen::VectorXd& new_value
   this->set_torques(new_values.segment(3 * this->get_size(), this->get_size()));
 }
 
-inline bool JointState::is_compatible(const State& state) const {
-  bool compatible = this->State::is_compatible(state);
-  compatible = compatible && (this->names_.size() == dynamic_cast<const JointState&>(state).names_.size());
+inline bool JointState::is_incompatible(const State& state) const {
+  bool compatible = (this->names_.size() == dynamic_cast<const JointState&>(state).names_.size());
   if (compatible) {
     for (unsigned int i = 0; i < this->names_.size(); ++i) {
       compatible = (compatible && this->names_[i] == dynamic_cast<const JointState&>(state).names_[i]);
     }
   }
```

Comment:
This can be simplified a bit, to avoid the dynamic cast in the for loop and streamline the AND operation.
```c++
auto other = dynamic_cast<const JointState&>(state);
bool compatible = (this->names_.size() == other.names_.size());
if (compatible) {
  for (unsigned int i = 0; i < this->names_.size(); ++i) {
    compatible &= this->names_[i] == other.names_[I];
  }
}
return !compatible;
```

But, I think early returns might just be simpler overall.
```c++
auto other = dynamic_cast<const JointState&>(state);
if (this->names_.size() != other.names_.size()) {
  return true;
}
for (unsigned int i = 0; i < this->names_.size(); ++i) {
  if (this->names_[i] != other.names_[I]) {
    return true;
  }
}
return false;
```